var/spack/repos/builtin/packages/abyss/package.py

```diff
@@ -0,0 +1,49 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Abyss(AutotoolsPackage):
+    """ABySS is a de novo, parallel, paired-end sequence assembler
+       that is designed for short reads. The single-processor version
+       is useful for assembling genomes up to 100 Mbases in size."""
+
+    homepage = "http://www.bcgsc.ca/platform/bioinfo/software/abyss"
+    url      = "http://www.bcgsc.ca/platform/bioinfo/software/abyss/releases/2.0.2/abyss-2.0.2.tar.gz"
+
+    version('2.0.2', '1623f55ad7f4586e80f6e74b1f27c798')
+
+    depends_on('openmpi', type='build')
```

Comment:
Oh, if you check the output of `configure --help`, you'll see that it does support MPICH (in Spack) and LAM (not in Spack) as well:
```
$ ./configure --help
...
  --enable-mpich          use MPICH (default is to use Open MPI)
  --enable-lammpi         use LAM/MPI (default is to use Open MPI)
```
You might want to change this to:
```python
depends_on('mpi')
```
and conflicts for the other MPI providers, and enable this flag `if 'mpich' in self.spec`.